### ISSUE TITLE
[BUG] conversions `from_nested_to_long` and `from_nested_to_multi_index` are broken

### ISSUE THREAD

**Issue by fkiraly** [sktime/sktime#2381]
`from_nested_to_long` and `from_nested_to_multi_index` are broken, they do not return containers in accordance with the specifications.
We should:
- [ ] fix this
- [ ] understand why the current tests in `datatypes` do not catch this (most likely lack of coverage in the `_examples.py` fixture set)
- [ ] ensure future tests catch this

To reproduce:

```python
from sktime.datasets import load_arrow_head
from sktime.datatypes._panel._convert import from_nested_to_long
X, y = load_arrow_head(return_X_y=True)
from_nested_to_long(X)
```
gives 
![image](https://user-images.githubusercontent.com/7985502/161441544-bfe0693d-a367-451f-8aed-d7d42b6d4359.png)

but it should be
![image](https://user-images.githubusercontent.com/7985502/161441547-13fb9120-8a2f-4185-9c76-a3bab61b0e43.png)

A straightforward fix was attempted here:
https://github.com/alan-turing-institute/sktime/pull/2375

but it seems to cause all kinds of strange ripple effects, especially in tsfresh and classifier modules.

Partial investigation indicates that the ripple effects may be caused by casting of `dtype` in `pandas` and `numpy`, where the wrong type may or may not cause things to break.

Even worse, the run https://github.com/alan-turing-institute/sktime/actions/runs/2085774075
seem to suggest that the problematic behaviour is likely python version and/or operating system specific.

**Comment by ltsaprounis** [comment]
> Partial investigation indicates that the ripple effects may be caused by casting of dtype in pandas and numpy, where the wrong type may or may not cause things to break.

Oh no! not again! #2224

**Comment by fkiraly** [comment]
yes indeed, well spotted!
It seems similar to yours!

**Comment by k1m190r** [comment]
```python
X.reset_index(drop=True)
```
fixes the result

The `X` from `load_arrow_head()` returning non-unique indexes which confuses `from_nested_to_long()`.

Should we reset in `from_nested_to_long()` or `raise ValuError`?

Definitely fix the  `load_arrow_head()` so it outputs only the unique indexes.

**Comment by fkiraly** [comment]
Hm, I see - the problem is a non-unique input.

What I think we should then do:
* we should check whether the checks check that the indices are unique
* the converters should raise a warning if non-unique indices are passed
* and do the `reset_index` fix inside the functions

Why: I think the longley data and conversions are used in a number of tests, estimators written by @TonyBagnall et al, and possibly by a number of users.
If we now let it raise `ValueError`, some of this code might break.
Instead, let's ensure it works with the broken input, and ensure that the `check` function would catch it.

**Comment by TonyBagnall** [comment]
we never user the longley data @fkiraly, nor long format converters at all. Also never use the merged data set, its why I changed the default return_X_y to true.

**Comment by fkiraly** [comment]
> we never user the longley data @fkiraly, nor long format converters at all. Also never use the merged data set, its why I changed the default `return_X_y` to true.

That's not right? It is used throughout `sktime`, for doc examples (which become tests via doctest) and some automated tests. Since it is one of the few on-board data sets that come multivariate.

The "original sin" here seems to be having created a data loader that, under certain circumstances, can return non-compliant data containers, but everyone will assume (since it is one of a few loaders) that the output is compliant, hence useable for testing!

**Comment by fkiraly** [comment]
fixed in #2375, forgot to close
